kingpin/actors/rightscale/base.py

```diff
@@ -213,6 +213,38 @@ def _delete_resource_tags(self, resource, tags):
         yield self._client.delete_resource_tags(resource, tags)
 
     @gen.coroutine
+    def _ensure_tags(self, res, tags):
+        """Ensures that a set of tags are applied to a RightScale resource.
+
+        Args:
+            res: The resource object itself
+            tags: A list of tags to apply
+        """
+        if isinstance(tags, basestring):
+            tags = [tags]
+
+        if not res.href:
```

Comment:
So the entire _ensure_tags() method will go away actually soon -- the existing `MCI` actor still uses it, but as soon as I have time to refactor that class to use the EnsurableBaseActor base, I'll get it cleaned up.